Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,271 @@
+  left_inv := fun L => fixedField_fixingSubgroup L
+  right_inv := by
+    intro H
+    simp_rw [fixingSubgroup_fixedField H]
+    rfl
+  map_rel_iff' := by
+    intro L₁ L₂
+    show L₁.fixingSubgroup ≥ L₂.fixingSubgroup ↔ L₁ ≤ L₂
+    rw [← fixedField_fixingSubgroup L₂, IntermediateField.le_iff_le, fixedField_fixingSubgroup L₂]
+
+/-- The Galois correspondence as a `GaloisInsertion` -/
+def GaloisInsertionIntermediateFieldClosedSubgroup [IsGalois k K] :
+    GaloisInsertion (OrderDual.toDual ∘ fun (E : IntermediateField k K) ↦
+      (⟨E.fixingSubgroup, fixingSubgroup_isClosed E⟩ : ClosedSubgroup (K ≃ₐ[k] K)))
+      ((fun (H : ClosedSubgroup (K ≃ₐ[k] K)) ↦ IntermediateField.fixedField H) ∘
+        OrderDual.toDual) :=
+  OrderIso.toGaloisInsertion IntermediateFieldEquivClosedSubgroup
+
+/-- The Galois correspondence as a `GaloisCoinsertion` -/
+def GaloisCoinsertionIntermediateFieldSubgroup [IsGalois k K] :
+    GaloisCoinsertion (OrderDual.toDual ∘ fun (E : IntermediateField k K) ↦ E.fixingSubgroup)
+      ((fun (H : Subgroup (K ≃ₐ[k] K)) ↦ IntermediateField.fixedField H) ∘
+        OrderDual.toDual) where
+  choice H _ := IntermediateField.fixedField H
+  gc E H := (IntermediateField.le_iff_le H E).symm
+  u_l_le K := le_of_eq (fixedField_fixingSubgroup K)
+  choice_eq _ _ := rfl
+
+theorem open_iff_finite (L : IntermediateField k K) [IsGalois k K] :
+    IsOpen (IntermediateFieldEquivClosedSubgroup L).carrier ↔
+    (FiniteDimensional k L) := by
+  refine ⟨fun h ↦ ?_, fun h ↦ IntermediateField.fixingSubgroup_isOpen L⟩
+  have : (IntermediateFieldEquivClosedSubgroup.toFun L).carrier ∈ nhds 1 :=
+    IsOpen.mem_nhds h (congrFun rfl)
+  rw [GroupFilterBasis.nhds_one_eq] at this
+  rcases this with ⟨S, ⟨gp, ⟨M, hM, eq'⟩, eq⟩, sub⟩
+  rw [← eq, ← eq'] at sub
+  have := hM.out
+  let L' : FiniteGaloisIntermediateField k K := {
+    normalClosure k M K with
+    finiteDimensional := normalClosure.is_finiteDimensional k M K
+    isGalois := IsGalois.normalClosure k M K }
+  have : L'.1.fixingSubgroup.carrier ⊆ (IntermediateFieldEquivClosedSubgroup.1.1 L).carrier := by
+    have : M ≤ L'.1 := IntermediateField.le_normalClosure M
+    rw [←  fixedField_fixingSubgroup L'.1, IntermediateField.le_iff_le] at this
+    exact fun _ a ↦ sub (this a)
+  apply IntermediateField.finiteDimensional_of_le (N := L'.1)
+  rw [← fixedField_fixingSubgroup L'.1, IntermediateField.le_iff_le]
+  exact this
+
+theorem normal_iff_isGalois (L : IntermediateField k K) [IsGalois k K] :
+    Subgroup.Normal (IntermediateFieldEquivClosedSubgroup L).1 ↔
+    IsGalois k L := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · let f : L → IntermediateField k K := fun x => IntermediateField.lift <|
+      IntermediateField.fixedField <| Subgroup.map (restrictNormalHom
+      (adjoin k {x.1})) L.fixingSubgroup
+    have h' (x : K) : (Subgroup.map (restrictNormalHom
+      (adjoin k {x})) L.fixingSubgroup).Normal := by
+      set f := restrictNormalHom (F := k) (K₁ := K) (adjoin k {x})
```

Comment:
Can't you pick another letter, since `f` has already been introduced? Also, use `let` instead of `set` if you don't need an equality inserted in the context.

Reply:
I think I can just insert it into the term, also needn't to add the `(F := k)` thing